src/compiler/checker.ts

```diff
@@ -5948,7 +5954,7 @@ namespace ts {
                 // If the current iteration type constituent is a string literal type, create a property.
                 // Otherwise, for type string create a string index signature.
                 if (t.flags & TypeFlags.StringLiteral) {
-                    const propName = escapeLeadingUnderscores((<StringLiteralType>t).value);

```

Comment:
Since this is a `StringLiteral`, does this need to be changed? Is it just for consistency?

Reply:
Mostly, yeah. The function below is used up above in the same body and is the correct one to use for future support of symbols as mapped type keys.